pkg/app/app.go

```diff
@@ -66,13 +67,16 @@ func CreateApp() App {
 		app.log.Error(err.Error())
 		os.Exit(1)
 	}
-	serv, clust, err := app.dao.GetEtcdVersion(app.config.Dao)
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	version, err := app.dao.Client.GetVersion(ctx)
```

Comment:
You could wrap this in a public method on dao.
```
func GetEtcdVersion(ctx...) {
return client.GetVersion(ctx)
}
```
That way you don't have to leak the client out only expose what you need to validate.